tox.ini

```
@@ -19,14 +19,15 @@ ignore=
 
 per-file-ignores=
     ; TYPE_CHECKING block suggestions
-    tests/*: TC001, TC002, TC003
+    tests/*: A, B, B904, C, E731, TC, ISC, N, SIM
```

I think the B904 is redundant?

```suggestion
    tests/*: A, B, C, E731, TC, ISC, N, SIM
```

---

No.

```bash
flake8 tests/
tests/conftest.py:160:21: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling.  See https://docs.python.org/3/tutorial/errors.html#exception-chaining for details.
tests/unit/test_links.py:73:17: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling.  See https://docs.python.org/3/tutorial/errors.html#exception-chaining for details.
tests/unit/test_terminal.py:171:17: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling.  See https://docs.python.org/3/tutorial/errors.html#exception-chaining for details.
```

---

Surely that means that the `B` exclusion isn't working then?

---

@oliver-sanders  `B` deals with all the other flake8 bugbear issues, just not this one. I've had a look at Flake8 Bugbear and can't find an issue. Having a quick play to see if I can make a toy example to report.

---

It's because we've explicitly added B904 to the `extend-select` section of `tox.ini`.